lbry/wallet/network.py

```diff
@@ -387,9 +398,16 @@ async def retriable_call(self, function, *args, **kwargs):
             try:
                 return await function(*args, **kwargs)
             except asyncio.TimeoutError:
-                log.warning("Wallet server call timed out, retrying.")
+                if self.exit_on_disconnect:
+                    log.error("Wallet server call timed out, exiting on server disconnect.")
+                    sys.exit(1)
+                else:
+                    log.warning("Wallet server call timed out, retrying.")
             except ConnectionError:
                 log.warning("connection error")
+                if self.exit_on_disconnect:
+                    log.error("exiting on server disconnect")
+                    sys.exit(1)
```

Comment:
You should consider adding SystemExit exception handling here: 
https://github.com/lbryio/lbry-sdk/blob/a334a93757d8e5c8310f7ab0bf74562a6d361993/lbry/extras/cli.py#L281

I guess the risk is that it would hang while shutting down components, but calling daemon.stop() is done in other scenarios like SIGINT, SIGTERM.